pyproject.toml

```diff
@@ -0,0 +1,112 @@
+[project]
+name = "emgcli"
+version = "2.4.22"
+readme = "README.md"
+authors = [
+  {name = "MGnify team", email = "metagenomics-help@ebi.ac.uk"},
+]
+license = {file = "LICENSE"}
+keywords = ["django", "api", "resource", "django-rest-framework", "jsonapi", "metagenomics"]
+
+dependencies = [
+    # deployment
+    "gunicorn==20.1.0",
+    "whitenoise==6.4.0",
+    "requests==2.28.1",
+    "yamjam==0.1.7",
+    "PyYAML==6.0",
+    # log handler
+    "concurrent-log-handler~=0.9.22",
+    # django
+    "Django==3.2.18",
+    "djangorestframework==3.12",
+    "django-filter==23.1",
+    "djangorestframework-jwt~=1.11.0",
+    "django-cors-headers==3.14.0",
+    "djangorestframework-jsonapi==4.2.1",
+    "djangorestframework-csv==2.1.1",
+    "drf-spectacular==0.26.0",
+    # ENA
+    "cx_Oracle~=6.2.1",
+    # mongo
+    "mongoengine==0.27.0",
+    "pymongo==4.3.3",
+    "django-rest-framework-mongoengine==3.4.1",
+    # my-sql
+    "django-mysql==4.3.0",
+    "mysqlclient==2.1.1",
+    "mysql-connector-python~=8.0.23",
+    "sqlparse==0.4.3",
+    # assembly contig viewer
+    "pysam==0.21.0",
+    # sourmash search
+    "celery[redis]==5.2.7",
+    # ena api lib
+    "ena_api_libs@git+https://github.com/EBI-Metagenomics/ena-api-handler.git@v2.0.2",
+]
+
+[project.urls]
+Homepage = "https://www.ebi.ac.uk/metagenomics"
+Documentation = "https://docs.mgnify.org"
+Repository = "https://github.com/ebi-metagenomics/emgapi"
+
+[build-system]
+requires = ["setuptools>=61.0.0"]
+build-backend = "setuptools.build_meta"
+requires-python = ">=3.8"
+
+[tool.setuptools.packages]
+find = {}
+
+[project.scripts]
+emgcli = "emgcli.manage:main"
+emgdeploy = "gunicorn.app.wsgiapp:run"
+
+[project.optional-dependencies]
+tests = [
+    "multidict==5.1.0",
+    "pytest==6.2.5",
+    "pytest-django==4.4.0",
+    "pytest-xdist==2.3.0",
+    "model_bakery==1.3.2",
+    "mock_services==0.3.1",
+    "mongomock==3.23.0",
+    "jsonapi-client==0.9.9",
+    "pytest-cov==2.12.1",
+    "pandas==1.3.2",
+    "responses==0.23.1",
+]
+
+dev = [
+    "django-debug-toolbar==3.8.1",
+    "django-extensions==3.2.1",
+    "django-grappelli==2.15.1"
+]
+
+admin = [
+    "django-grappelli==2.15.1",
+]
+
+webuploader = [
+    "pandas==1.3.2"
+]
+
+[tool.pytest.ini_options]
+addopts = "-p no:warnings --cov-report term --cov=emgapi --cov=emgapianns --cov=emgcli --cov=emgena"
+python_files = "tests/*.py"
+
+[tool.tox]
+legacy_tox_ini = """
+[flake8]
```

Comment:
https://github.com/charliermarsh/ruff is faster than flake8.. should we migrate to it at some point